lib/generic-request.js

```diff
@@ -0,0 +1,27 @@
+const isCorrectHttpMethod = require('./check-http-method');
+const HTTP_STATUS_CODES = require('./http-status-codes');
+const rxjs = require('rxjs');
+const Observable = rxjs.Observable;
+const throwError = rxjs.throwError;
+const request = require('request');
+const isFailedResponse = statusCode => statusCode >= HTTP_STATUS_CODES.BAD_REQUEST;
+
+module.exports = (method, url, options) => {
+    if (!isCorrectHttpMethod(method)) {
+        const errMsg = `httpRx client was given an invalid method name for http request: ${method}`;
+        return throwError(new Error(errMsg));
+    }
+    return new Observable(observer => {
```

Comment:
I think that is considered better to use Observable.create instead of new operator

Reply:
I am fine with any style as long as we get the same result.
Will do.

Reply:
Replaced with static `Observable.create`.